common/djangoapps/student/tests/test_helpers.py

```diff
@@ -67,6 +67,8 @@ def test_unsafe_next(self, log_level, log_name, unsafe_url, http_accept, user_ag
     @ddt.data(
         ('/dashboard', 'testserver', '/dashboard'),
         ('https://edx.org/courses', 'edx.org', 'https://edx.org/courses'),
+        ('https://test.edx.org/courses', 'edx.org', 'https://test.edx.org/courses'),
+        ('https://test.edx.org/courses', 'courses.edx.org', 'https://test.edx.org/courses'),
```

Comment:
@nasthagiri: FYI: I added this test and the 'courses.edx.org' comment above just to ensure that this works as planned before merge.

Additionally, please read the opening PR comment because I had some questions there.  I'm guessing this can still merge, but maybe we'd want to document refactor work for later?  Maybe it is just obvious?